libs/community/langchain_community/document_loaders/recursive_url_loader.py

```diff
@@ -169,6 +169,7 @@ def _get_child_links_recursive(
         visited.add(url)
         try:
             response = requests.get(url, timeout=self.timeout, headers=self.headers)
+            response.encoding = response.apparent_encoding
```

Comment:
I don't think that apparent encoding is bullet proof. It's an inferred encoding and the inferred encoding can be incorrect while the actual encoding could be correct. Is that not the case?

It looks like the logic needs to be a bit more complex or opt-in rather than the default?

Reply:
Maybe we can add arguments `encoding` and `autoset_encoding` just like `WebBaseLoader`.
https://github.com/langchain-ai/langchain/blob/940242c1ecacfa85a5de85a261c68bb23174bdad/libs/community/langchain_community/document_loaders/web_base.py#L49-L50

If `encoding` value is given by users, we use the given one. Otherwise, if `autoset_encoding` is `True` (by default), we use apparent encoding.
https://github.com/langchain-ai/langchain/blob/940242c1ecacfa85a5de85a261c68bb23174bdad/libs/community/langchain_community/document_loaders/web_base.py#L232-L235

How does it sound like?